### PR TITLE
OCPBUGS-22824: Fix error handling for Approve()

### PR DESCRIPTION
**Pull request by sebsoto**
This commit ensures valid and err are being handled appropriately. The logic was mistakenly changed in https://github.com/openshift/windows-machine-config-operator/commit/8b7275ab24d74a55463aa0f090cfec39dc358f42

### CODE REVIEW

File: pkg/csr/csr.go [hunk 1]
```diff
@@ -100,16 +100,19 @@ func NewApprover(client client.Client, clientSet *kubernetes.Clientset, csr *cer
...
 func (a *Approver) Approve() error {
 	if a.k8sclientset == nil {
 		return fmt.Errorf("kubernetes clientSet should not be nil")
 	}
 
-	if valid, err := a.validateCSRContents(); !valid && err != nil {
-		// if the validation fails and error returned is nil, returns nil
-		return fmt.Errorf("could not validate contents for approval of CSR: %s: %w", a.csr.Name, err)
-
+	shouldBeApproved, err := a.validateCSRContents()
```
Comment (jrvaldes) [inline pkg/csr/csr.go:110]:
```suggestion
	validForApproval, err := a.validateCSRContents()
```

To align with the function' name.

**Review by jrvaldes — COMMENTED** [review]
@sebsoto  thanks for fixing this so quickly. LGTM, just a minor suggestion. 

Does it make sense to add a unit test to ensure the proposed workflow?

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Review by aravindhp — APPROVED** [review]
/lgtm

**Pull request merged**